### ISSUE TITLE
Cloud Data Sync in Progress (Error 77)

### ISSUE THREAD

**Issue by ttgolf** [nfarina/homebridge#348]
I am hoping someone may be able to help me.  I just upgraded my phone to iPhone 6S Plus.  After restore from my backup, I no longer can work with any of the apps (iDevices, Insteon, MyTouchHome, etc) due to the "Cloud Data Sync in Progress" error code.  

I have tried delete/reinstall apps, resetting the iCloud Keychain and log in and out of my iCloud account multiple times without any success.  Note I spent many hours last night logging out of iCloud from all my iPhones, iPad and Macs.  iCloud keychain was also reset.  I even went to iCloud.com and make sure I delete out the list of all my devices.  It still does not work even when I only have one device log into iCloud.

The second thing I tried was creating a new Apple ID and invite myself (with my existing Apple ID).  The good news is that I can setup Homebridge with the new AppleID.  The bad news is I don't even receive invitations from the new AppleID account.  Therefore this workaround does not work. I know the invitation does go out because my wife and my son both get the invitations on their iPhones.  

The third thing I did this morning was calling Apple Support.  As expected Apple does not know how to resolve this issue.  

The REALLY bad news - As a result of me logging in and out with all my devices, I now get the same error message with all my iPads and iPhones.  Basically the only way I can use my home automation is to login with a new Apple ID.  Please help!!!!

**Comment by KhaosT** [comment]
Have you tried to log-in with your old Apple ID and perform a reset HomeKit under Settings > Privacy > HomeKit ?

**Comment by ttgolf** [comment]
Yes multiple times on all my iOS devices as well.

**Comment by KhaosT** [comment]
That's interesting… what about sign out Apple ID from the device, reboot, open a HomeKit app, create a home and then sign back in Apple ID?

**Comment by ttgolf** [comment]
I tried that too.  If I am out of AppleID.  I have no problem creating a home and everything works.  But if I sign back into AppleID, I won't be able to do anything (add, delete or modify with my home).  Every change I try to make will come back with the Cloud Data Sync in Progress error code.

**Comment by ttgolf** [comment]
I called iDevices support today about the error code (since I have one of their smart switch).  The tech support guy is about 100 times more technical than the Apple support.  They told me they are seeing a lot of people with similar issues.  They are telling their customers to log out and log back into iCloud.  The tech support guy says they have informed Apple about this issue.  But they really don't know what goes on with iCloud.  They are troubleshooting this problem in the "dark".

**Comment by KhaosT** [comment]
Yep it's impossible to debug this user from user side... I'd suggest to wait a few days and give it another try...

**Comment by ttgolf** [comment]
Thanks for your help.  Will try in a few more days

**Comment by ttgolf** [comment]
OK.  Finally fixed the problem by accident.  So for anyone who has similar issue I hope this might help.

1.  Log out of iCloud account
2.  Reset Homekit under Settings/Privacy/Homekit
3.  REBOOT (most of what you read online skipped this step.  I found out without rooting it won't work)
4.  Log back into iCloud account
5.  Setup Homebridge (note you might have to delete the persist folder and start over.

@KhaosT Thanks for your help.

**Comment by julianfez** [comment]
I used to have this problem and I talked with Apple and it's very simple I did just reset network settings and this immediately helps , the problem that I have if someone could help me is inviting for share HomeKit, received the invitation and accept but never show up the home on any app just in HomeKit in settings, anyone know how to fix this?

**Comment by ttgolf** [comment]
Reset network setting did not work for me.  But it is good to know for anyone else trying to troubleshoot this problem.

**Comment by rafale77** [comment]
Very odd... I have 4 iOS devices, All had the my homekit configured properly up until today. 3 of them dropped off and I am left only with one still having all the information. This is the first time it is occurring. I would have unreachable devices before but this is the entire homekit gone from 3 devices out of 4.

**Comment by ObieOne1** [comment]
Just to share, but I have the exact same issue. When logged-out of iCloud all works normally, but as soon as I login to iCloud i get the message 'iCloud data sync is in progress...' and homekit does not work.
Even after a days of waiting, still the same message.

Tried the above steps, but it does not work a bit. So for now, no Siri to control the house...

**Comment by julianfez** [comment]
Also try reset homekit and add new home , that should work ,

**Comment by jpawlowski** [comment]
Some issue here.

It is most likely an issue with Apple's Homekit framework itself. I just filed a bug report using the Feedback.app in iOS9.2beta. I suggest you all do the same to increase the priority. This can be done here as well: https://bugreport.apple.com

**Comment by rafale77** [comment]
The way it works is a mystery. I had one ipad with all the configs correct which could never sync with none of the other 3 devices. I decided to restart from scratch. Adding devices using my phone would sync instantly to one of my ipads but not to the other 2. That's until last night when one of the other ipads suddenly would sync after I reboot it but I still have one more which doesn't work. They are all on the same iOS version and all running the same iCloud settings.

**Comment by timc166293** [comment]
Great thread. 

Found fix. Reset network setting under settings/ general. Also fixed slow web page loading. Crazy:). 

Good luck.

**Comment by WaynerSage** [comment]
You may want to think twice about removing your iCloud account as it may not solve this problem but it seems to delete all of your Apple Pay cards.

**Comment by jsm1234** [comment]
I had the same issue of not being able to access homekit via my insteon app due to the sync being in progress.  I let it sync for a full 24 hours and it didn't work.  First I did the above (ttgolf's steps 1-5) and it didn't work.  I also signed out of all my icloud devices and I added in the reset network step suggested.  That didn't work either.  But, after lots of frustration .. I finally got it to work.  

I read around on the internet, and the gist was that something was holding up the sync.  So I added two steps. First, after ttgolf's step 1 I also signed out of iTunes on the iTunes app. Second, after ttgolf's step4,  when I signed back into iCloud, I basically turned off all of the options except keychain (Settings: iCloud, then all the toggle buttons including drive and photos i turned to off). 

After that, homekit and the insteon app worked just fine.  And one by one, i un-toggled all my icloud settings, and then signed back into itunes.

Last but not least I would add ... I would DEFINITELY backup all your photos, music etc before doing this.  I did it just in case ... and I didn't need it, but ... always a good idea.
